dsgrid/utils/scratch_dir_context.py

```diff
@@ -0,0 +1,49 @@
+import os
+import shutil
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+
+
+class ScratchDirContext:
+    """Manages the lifetime of files in a scratch directory."""
+
+    def __init__(self, scratch_dir: Path):
+        self._scratch_dir = scratch_dir
+        self._paths: list[Path] = []
+        if not self._scratch_dir.exists():
+            self._scratch_dir.mkdir()
+
+    @property
+    def scratch_dir(self) -> Path:
+        """Return the scratch directory."""
+        return self._scratch_dir
+
+    def add_tracked_path(self, path: Path) -> None:
+        """Add tracking of a path in the scratch directory."""
+        self._paths.append(path)
+
+    def list_tracked_paths(self) -> list[Path]:
+        """Return a list of paths being tracked."""
+        return self._paths[:]
```

Comment:
That is a loaded question. :) I clearly prefer what I wrote. It says, “return a full slice (not a copy) of the list.” It is equivalent to calling `list`. Is calling `list` more clear to others?

Reply:
I kind of understood what [:] should do, but I still looked it up because it was unfamiliar to me ... Not sure how @lixiliu or @ashreeta feel about it.

Reply:
Same, not familiar with the [:] notation but the output type hint says it's a list, so that's clear. I don't have a preference here.